redash/handlers/users.py

```diff
@@ -27,7 +27,7 @@ def get(self):
     def post(self):
         req = request.get_json(force=True)
         require_fields(req, ('name', 'email'))
-
+        req['email'] = req['email'].lower()
```

Comment:
I agree. Perhaps updating the [User](https://github.com/getredash/redash/blob/master/redash/models.py#L318) model and listen for a set event on the email field, would be better. Then just make sure it is lowercase: 
```
@listens_for(User.email, 'set')
def make_email_lowercase(target, val, oldval, initiator):
    target.email = val.lower()
```

_Note: Haven't tested the code._